### DISCUSSION TITLE
How to forward docker.sock

### DISCUSSION THREAD

**Discussion by elyrtan** [abiosoft/colima#688] [Q&A] [answered]
Hi,

i'm playing around with gitlab runner docker image and trying to have it talk with the docker host engine.
I run the container like so: 
```shell
docker run -d --name gitlab-runner --restart always \
-v gitlab-runner-config:/etc/gitlab-runner \
-v /Users/myUser/.colima/default/docker.sock:/var/run/docker.sock \
gitlab/gitlab-runner:latest
``` 

However the runner looks unable to talk with the docker engine for now, ci jobs are failing with :
```shell
ERROR: Preparation failed: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running? (docker.go:754:0s)
Will be retried in 3s ...
ERROR: Failed to remove network for build
ERROR: Preparation failed: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running? (docker.go:[7](https://gitlab.com/blah/hello-world/-/jobs/4069024169#L7)54:0s)
``` 

Just to make sure, I've tried running portainer container which looks unable to talk with the docker host engine as well.

Is there something configurable colima side ?

My colima status:
```shell
INFO[0000] colima is running using QEMU
INFO[0000] arch: aarch64
INFO[0000] runtime: docker
INFO[0000] mountType: sshfs
INFO[0000] address: 192.168.106.2
INFO[0000] socket: unix:///Users/myUser/.colima/default/docker.sock
```

**Comment by jandubois** [comment, marked as answer]
> ```shell
> -v /Users/myUser/.colima/default/docker.sock:/var/run/docker.sock \
> ```

This is the location of the socket on the host. You need to forward the location inside the VM, so probably:

```
-v /var/run/docker.sock:/var/run/docker.sock
```

**Reply by elyrtan** [reply]
Why didnt I think of that ...
Thank you !

**Reply by EduPonz** [reply]
This just saved my day!

**Reply by jubr** [reply]
Saved my hide too for a Docker-in-docker use case involving Home Assistant addon building.

Perhaps it would make sense to add this to the FAQ, @abiosoft? What do you think?
